store/tikv/error.go

```diff
@@ -44,12 +44,13 @@ var (
 
 func init() {
 	tikvMySQLErrCodes := map[terror.ErrCode]uint16{
-		mysql.ErrTiKVServerTimeout:  mysql.ErrTiKVServerTimeout,
-		mysql.ErrResolveLockTimeout: mysql.ErrResolveLockTimeout,
-		mysql.ErrPDServerTimeout:    mysql.ErrPDServerTimeout,
-		mysql.ErrRegionUnavailable:  mysql.ErrRegionUnavailable,
-		mysql.ErrTiKVServerBusy:     mysql.ErrTiKVServerBusy,
-		mysql.ErrGCTooEarly:         mysql.ErrGCTooEarly,
+		mysql.ErrTiKVServerTimeout:   mysql.ErrTiKVServerTimeout,
+		mysql.ErrResolveLockTimeout:  mysql.ErrResolveLockTimeout,
+		mysql.ErrPDServerTimeout:     mysql.ErrPDServerTimeout,
+		mysql.ErrRegionUnavailable:   mysql.ErrRegionUnavailable,
+		mysql.ErrTiKVServerBusy:      mysql.ErrTiKVServerBusy,
+		mysql.ErrGCTooEarly:          mysql.ErrGCTooEarly,
+		mysql.ErrTruncatedWrongValue: mysql.ErrTruncatedWrongValue,
```

Comment:
Why add this error? It isn't used.

Reply:
It's used by expression evaluation. We have to add this error code mapping to avoid `mysql.ErrTruncatedWrongValue` being treated as `defaultMySQLErrorCode`, witch is `ErrUnknown`(1105)

Reply:
@AndreMouche PTAL